src/Data/TypeRepMap/Internal.hs

```diff
@@ -74,6 +75,9 @@ data TypeRepMap (f :: k -> Type) =
     }
   -- ^ an unsafe constructor for 'TypeRepMap'
 
+instance NFData (TypeRepMap f) where
```

Comment:
`deepseq` is one of the libraries that go with GHC itself. So I'm not sure how much is that a problem. Especially if users may want to run `rnf` on your structure.

Reply:
@qnikst Well, in dependencies you usually specify range of versions. So it's still possible that GHC will build some library under certain constraints. So I think that it's still a good idea to keep small dependency footprint.

Regarding calling `rnf` on our structure: well, I don't see this happening. When it happens, the person can open issue and we will weight all pros and cons and decide what to do.

Reply:
Please check the type of `Criterion.Main.env`

Reply:
@qnikst I've checked the type of the `env` function. I just think that `NFData` instance can be orphan instance inside benchmarks instead of having extra dependency in the library.

Reply:
Since `deepseq` is shipped with GHC, I agree with @qnikst that it's better to have `NFData` in the library, not in the benchmarks.

I see no reason for `TypeRepMap` to *not* have an `NFData` instance.

Reply:
Okay, let's keep it then